llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -3013,8 +3014,11 @@ class VPExpressionRecipe : public VPSingleDefRecipe {
                            {Ext0, Ext1, Mul, Red}) {}
 
   ~VPExpressionRecipe() override {
-    for (auto *R : reverse(ExpressionRecipes))
-      delete R;
+    SmallSet<VPSingleDefRecipe *, 4> ExpressionRecipesSeen;
+    for (auto *R : reverse(ExpressionRecipes)) {
+      if (ExpressionRecipesSeen.insert(R).second)
```

Comment:
Can you update the comment for `SmallVector<VPSingleDefRecipe *> ExpressionRecipes` (line 2973) that this vector may contain duplicates?